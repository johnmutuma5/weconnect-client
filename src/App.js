import React, { Component } from 'react';
import Layout from './containers/Layout/Layout';
import { Route } from 'react-router-dom';
import { Switch } from 'react-router';
import Businesses from './containers/Businesses/Businesses';
import BusinessesFilter from './containers/BusinessesFilter/BusinessesFilter';
import SearchBusinesses from './containers/Search/Search';
import UserRegistrationForm from './containers/Auth/UserRegistration/UserRegistrationForm';
import BusinessProfile from './containers/BusinessProfile/BusinessProfile';
import ResourcesProvider, { createStore } from './store';
import weConnectReducer from './store/baseReducer';


export const store = createStore(weConnectReducer);

class App extends Component {
  render() {
    return (
        <ResourcesProvider store={ store }>
            <Layout>
                <Switch>
                    <Route exact path="/" component={ Businesses } />
                    <Route exact path="/businesses" component={ Businesses } />
                    <Route exact path="/businesses/filter" component={ BusinessesFilter } />
                    <Route exact path="/businesses/search" component={ SearchBusinesses } />
                    <Route exact path="/businesses/:id" component={ BusinessProfile } />
                </Switch>
            </Layout>
        </ResourcesProvider>
    );
  }
}

export default App;
