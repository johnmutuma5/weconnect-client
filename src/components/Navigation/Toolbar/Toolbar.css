.toolbar {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  height: 50px;
  width: 100%;
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
  background-color: #00396D;
  padding: 0 15px;
  position: fixed;
  top: 0;
  z-index: 99;
  margin-bottom: 10px;
}

.toolbar::after {
    content: "";
    -webkit-box-shadow: 3px 2px 2px 0px #00080C;
            box-shadow: 3px 2px 2px 0px #00080C;
    position: absolute;
    bottom: 0;
    left: 0;
    width: 100%;
    height: 3px;
}
