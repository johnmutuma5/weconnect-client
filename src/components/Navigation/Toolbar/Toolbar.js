import React from 'react';
import PropTypes from 'prop-types';
import NavigationItems from '../NavigationItems/NavigationItems';
import Logo from '../../Logo/Logo';
import SideMenuToggler from '../../UI/SideMenuToggler/SideMenuToggler';
import Button from '../../UI/Button/Button';
import logo from '../../../Logo.svg';
import { toggleGettingStarted } from '../../../store/actions/actions';

import './Toolbar.css';

const toolbar = (props, context) => {
    const store = context.store;
    const layoutState = context.layoutState;

    let gettingStartedButton =
        <Button onClickHandler={ () => store.dispatch(toggleGettingStarted()) }>
            { 'Get Started' }
        </Button>;

    // check layout state
    if (layoutState.showLayoutForAuthenticatedUser)
        gettingStartedButton = 'User Avatar'; // to import from profile components

    return (
        <div className="Toolbar container">
            <Logo src= { logo } alt_text="WeConnect" />
            <NavigationItems />
            { gettingStartedButton }
            <SideMenuToggler click={ props.click } />

        </div>
    );
};

toolbar.contextTypes = {
    store: PropTypes.object.isRequired,
    layoutState: PropTypes.object.isRequired
}

export default toolbar;
