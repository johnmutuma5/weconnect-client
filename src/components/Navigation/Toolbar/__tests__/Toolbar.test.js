import MemoryRouter from 'react-router-dom';
import Toolbar from '../Toolbar';
import { store } from '../../../../App'

describe('<Toolbar /> Component', () => {
    let context;

    beforeEach(() => {

        store.state = {
            authState: { userToken: undefined },
            layoutState: {}
        };
        context = {
            store
        }
    });

    it('renders correctly', () => {
        const wrapper = shallow(<Toolbar />, { context });
        expect(wrapper).toMatchSnapshot();
    });

    it('displays logout button when authenticated', () => {
        expect.assertions(2);
        context.store.state.authState = { userToken: 'a.user.token'};
        context.store.state.layoutState = {showLayoutForAuthenticatedUser: true}

        const wrapper = shallow(<Toolbar />, { context });
        expect(wrapper).toMatchSnapshot();

        const buttonText = wrapper.find('button').dive().text();
        expect(buttonText).toContain('Logout');
    });
})
