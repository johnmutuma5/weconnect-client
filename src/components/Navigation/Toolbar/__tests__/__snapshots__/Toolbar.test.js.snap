// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Toolbar /> Component displays logout button when authenticated 1`] = `
<div
  className="Toolbar container"
>
  <div
    className="Logo"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <div>
        <img
          alt="WeConnect"
          src="Logo.svg"
        />
      </div>
    </a>
    <div
      className="synopsis"
    >
      <p>
        Put your businesses on a global platform...
      </p>
      <p>
        Looking to buy? You can find other businesses in your location
      </p>
    </div>
  </div>
  <nav
    className="NavigationItems"
  >
    <ul />
  </nav>
  <article
    className="Search hidden"
  >
    <div
      className="SearchIcon"
      onClick={[Function]}
    >
      <span />
      <span />
    </div>
    <div
      className="form_container"
    >
      <form
        onSubmit={[Function]}
      >
        <input
          key="searchValue"
          name="searchValue"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Search weConnect"
          type="text"
          value=""
        />
      </form>
    </div>
  </article>
  <div
    className="ProfileTools"
    id="atToolBar"
  >
    <div
      className="ProfileAvatar"
    >
      <div
        className="avatar-img"
      >
        <img
          alt="Pic"
          src="Avatar.svg"
        />
      </div>
    </div>
    <div
      className="ProfileItems"
    >
      <div />
      <div
        className="ProfileActions"
      >
        <div
          className="Button dark"
          onClick={[Function]}
        >
           
          Logout
        </div>
      </div>
    </div>
  </div>
  <div
    className="SideMenuToggler visible"
    onClick={[Function]}
  >
    <span />
    <span />
    <span />
  </div>
</div>
`;

exports[`<Toolbar /> Component renders correctly 1`] = `
<div
  className="Toolbar container"
>
  <logo
    alt_text="WeConnect"
    src="Logo.svg"
  />
  <withRouter(navigationItems) />
  <withRouter(SearchField) />
  <Component
    id="toolbarGetStarted"
  />
  <sideMenuToggler />
</div>
`;
